File: university-timetable/src/types/timetable.ts
export interface Timeslot {
  id: string;
  dayOfWeek: string;
  startTime: string;
  endTime: string;
}

export interface Room {
  id: string;
  name: string;
}

export interface StudentGroup {
  id: string;
  name: string;
}

export interface Teacher {
  id: string;
  name: string;
}

export interface Lesson {
  id: string;
  subject: string;
  teacher: Teacher;
  studentGroup: StudentGroup;
  timeslot?: Timeslot;
  room?: Room;
}

export interface Timetable {
  name: string;
  timeslots: Timeslot[];
  rooms: Room[];
  studentGroups: StudentGroup[];
  teachers: Teacher[];
  lessons: Lesson[];
  score?: {
    hardScore: number;
    softScore: number;
  };
  solverStatus?: string;
}

File: getting-started/quarkus-console/src/main/java/org/acme/schooltimetabling/domain/Timetable.java
package org.acme.schooltimetabling.domain;

import java.util.List;

import ai.timefold.solver.core.api.domain.solution.PlanningEntityCollectionProperty;
import ai.timefold.solver.core.api.domain.solution.PlanningScore;
import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
import ai.timefold.solver.core.api.domain.solution.ProblemFactCollectionProperty;
import ai.timefold.solver.core.api.domain.valuerange.ValueRangeProvider;
import ai.timefold.solver.core.api.score.HardSoftScore;

@PlanningSolution
public class Timetable {

    private String name;

    @ProblemFactCollectionProperty
    @ValueRangeProvider
    private List<Timeslot> timeslots;
    @ProblemFactCollectionProperty
    @ValueRangeProvider
    private List<Room> rooms;
    @PlanningEntityCollectionProperty
    private List<Lesson> lessons;

    @PlanningScore
    private HardSoftScore score;

    // No-arg constructor required for Timefold
    public Timetable() {
    }

    public Timetable(String name, List<Timeslot> timeslots, List<Room> rooms, List<Lesson> lessons) {
        this.name = name;
        this.timeslots = timeslots;
        this.rooms = rooms;
        this.lessons = lessons;
    }

    // ************************************************************************
    // Getters and setters
    // ************************************************************************

    public String getName() {
        return name;
    }

    public List<Timeslot> getTimeslots() {
        return timeslots;
    }

    public List<Room> getRooms() {
        return rooms;
    }

    public List<Lesson> getLessons() {
        return lessons;
    }

    public HardSoftScore getScore() {
        return score;
    }

    public void setScore(HardSoftScore score) {
        this.score = score;
    }

}
